syntaxes/src/expression/grammar.ts

```diff
@@ -0,0 +1,886 @@
+          match: /(?<!\.|\$)\bdelete\b(?!\$)/,
+        },
+        {
+          name: 'keyword.operator.expression.in.ts',
+          match: /(?<!\.|\$)\bin\b(?!\$)/,
+        },
+        {
+          name: 'keyword.operator.expression.of.ts',
+          match: /(?<!\.|\$)\bof\b(?!\$)/,
+        },
+        {
+          name: 'keyword.control.if.ts',
+          match: /(?<!\.|\$)\bif\b(?!\$)/,
+        },
+        {
+          name: 'keyword.control.else.ts',
+          match: /(?<!\.|\$)\belse\b(?!\$)/,
+        },
+        {
+          name: 'keyword.control.then.ts',
+          match: /(?<!\.|\$)\bthen\b(?!\$)/,
+        },
+        {
+          name: 'keyword.operator.expression.instanceof.ts',
+          match: /(?<!\.|\$)\binstanceof\b(?!\$)/,
+        },
+        {
+          name: 'keyword.operator.new.ts',
+          match: /(?<!\.|\$)\bnew\b(?!\$)/,
+        },
+        {
+          include: '#typeofOperator',
+        },
+        {
+          name: 'keyword.operator.expression.void.ts',
+          match: /(?<!\.|\$)\bvoid\b(?!\$)/,
+        },
+        {
+          begin: /(?<!\.|\$)\bas\b(?!\$)/,
+          beginCaptures: {
+            '0': {
+              name: 'keyword.control.as.ts',
+            },
+          },
+          end: /(?=$|"|[;,:})\]])/,
+          patterns: [
+            {
+              include: '#type',
+            },
+          ],
+        },
+        {
+          name: 'keyword.operator.assignment.compound.ts',
+          match: /\*=|(?<!\()\/=|%=|\+=|\-=/,
+        },
+        {
+          name: 'keyword.operator.assignment.compound.bitwise.ts',
+          match: /\&=|\^=|<<=|>>=|>>>=|\|=/,
+        },
+        {
+          name: 'keyword.operator.bitwise.shift.ts',
+          match: /<<|>>>|>>/,
+        },
+        {
+          name: 'keyword.operator.comparison.ts',
+          match: /===|!==|==|!=/,
+        },
+        {
+          name: 'keyword.operator.relational.ts',
+          match: /<=|>=|<>|<|>/,
+        },
+        {
+          name: 'keyword.operator.logical.ts',
+          match: /\!|&&|\|\|/,
+        },
+        {
+          name: 'keyword.operator.bitwise.ts',
+          match: /\&|~|\^|\|/,
+        },
+        {
+          name: 'keyword.operator.assignment.ts',
+          match: /\=/,
+        },
+        {
+          name: 'keyword.operator.decrement.ts',
+          match: /--/,
+        },
+        {
+          name: 'keyword.operator.increment.ts',
+          match: /\+\+/,
+        },
+        {
+          name: 'keyword.operator.arithmetic.ts',
+          match: /\%|\*|\/|-|\+/,
+        },
+        {
+          match: /(?<=[_$[:alnum:]])\s*(\/)(?![\/*])/,
+          captures: {
+            '1': {
+              name: 'keyword.operator.arithmetic.ts',
+            },
+          },
+        },
+      ],
+    },
+
+    functionCall: {
+      begin: /(?=(\??\.\s*)?([_$[:alpha:]][_$[:alnum:]]*)\s*(<([^<>]|\<[^<>]+\>)+>\s*)?\()/,
```

Comment:
Will `[:alpha:]` match all alpha characters or only :, a, l, p, h?

Reply:
`[:alpha:]` is a POXIS bracket used in Oniguruma, the vscode regex engine. This is the same as `[a-zA-Z]`

Reply:
wow. cool.